app/models/payments/OpenPaymentsModel.scala

```diff
@@ -18,10 +18,148 @@ package models.payments
+    case FtnEachPartnerCharge => messages("openPayments.vatFtnEachpartner")
+    case MpPre2009Charge => messages("openPayments.vatMpPre2009")
+    case MpRepeatedPre2009Charge => messages("openPayments.vatMpRepeatedPre2009")
+    case CivilEvasionPenaltyCharge => messages("openPayments.vatCivilEvasionPenalty")
+  }
+
+  def makePaymentRedirect: String
+}
+
+object OpenPaymentsModel {
+
+  def apply(chargeType: ChargeType,
+            amount: BigDecimal,
+            due: LocalDate,
+            start: LocalDate,
+            end: LocalDate,
+            periodKey: String,
+            overdue: Boolean): OpenPaymentsModel = OpenPaymentsModelWithPeriod(chargeType, amount, due, start, end, periodKey, overdue)
+
+  def apply(chargeType: ChargeType,
+            amount: BigDecimal,
+            due: LocalDate,
+            periodKey: String,
+            overdue: Boolean): OpenPaymentsModel = OpenPaymentsModelNoPeriod(chargeType, amount, due, periodKey, overdue)
+
+  def apply(payment: Payment, overdue: Boolean)(implicit messages: Messages): OpenPaymentsModel = payment match {
+    case payment: PaymentWithPeriod => OpenPaymentsModelWithPeriod(
+      payment.chargeType,
+      payment.outstandingAmount,
+      payment.due,
+      payment.start,
+      payment.end,
+      payment.periodKey,
+      overdue
+    )
+    case payment: PaymentNoPeriod => OpenPaymentsModelNoPeriod(
+      payment.chargeType,
+      payment.outstandingAmount,
+      payment.due,
+      payment.periodKey,
+      overdue
+    )
+  }
+
+  implicit val writes: Writes[OpenPaymentsModel] = Writes {
+    case model: OpenPaymentsModelWithPeriod => Json.toJson(model)(OpenPaymentsModelWithPeriod.writes)
+    case model: OpenPaymentsModelNoPeriod => Json.toJson(model)(OpenPaymentsModelNoPeriod.writes)
+  }
+}
+
+case class OpenPaymentsModelWithPeriod(chargeType: ChargeType,
+                                       amount: BigDecimal,
+                                       due: LocalDate,
+                                       start: LocalDate,
+                                       end: LocalDate,
+                                       periodKey: String,
+                                       overdue: Boolean = false) extends OpenPaymentsModel {
+
+  override def whatYouOweDescription(implicit messages: Messages): String = chargeType match {
+    case ReturnDebitCharge => messages("openPayments.vatReturn", displayDateRange(start, end))
+    case DefaultSurcharge => messages("openPayments.surcharge", displayDateRange(start, end))
+    case CentralAssessmentCharge => s"${
+      messages("openPayments.centralAssessment", displayDateRange(start, end)).trim}${
+      messages("openPayments.centralAssessmentSubmit")
+    }"
+    case ErrorCorrectionDebitCharge => messages("openPayments.errorCorrection", displayDateRange(start, end)).trim
+    case AAInterestCharge =>
+      s"${messages.apply("openPayments.AADefaultInterestDescription",{displayDateRange(start, end)}).trim}"
```

Comment:
You don't need the string concatenation here and a lot of `{ ... }` can be removed in this def in general